src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -73,9 +73,11 @@ void InterceptBallAction::calculateNextIntent(IntentCoroutine::push_type& yield)
         bool point_in_front_of_ball =
             pointInFrontVector(ball.position(), ball.velocity(), closest_point);
 
-        // We add 1e-6 to avoid division by 0 without affecting the result significantly
-        Duration ball_time_to_position = Duration::fromSeconds(
-            distance(closest_point, ball.position()) / (ball.velocity().length() + 1e-6));
+        // We add FIXED_EPSILON to avoid division by 0 without affecting the result
+        // significantly
+        Duration ball_time_to_position =
```

Comment:
This may be a case where `1e-6` is preferred over `FIXED_EPSILON`? Although they technically achieve the same thing here, FIXED_EPSILON is so small I wouldn't be as confident we wouldn't blow up the result of the division (perhaps to the point of overflow?). `1e-6` seems to be reasonably chosen since a `micrometer/s` wont' affect the behavior of the calculation but is more "numerically stable" at preventing division by 0.

@garethellis0 for numerical stability, @jonathanlew @akhilveeraghanta for opinions

Reply:
What about an explicit `ball.velocity().length()!=0` check?

Reply:
Yeah I vouch for Jons point, adding epsilon to avoid dividing by zero almost seems like a hack imo, much rather prefer a check to make sure the divisor is not zero.

Reply:
So adding a fixed epsilon the denominator like this is _fairly_ common practice, but I agree with Jon and Akhil here, adding an explicit check seems like a much more reasonable option in this case.

Reply:
So, what do we do if the `ball.velocity().length` gives zero? Do we just immediately return?

Reply:
Added check.